src/charm.py

```diff
@@ -87,6 +93,14 @@ def container(self) -> Container:
         container = self.unit.get_container(FLASK_CONTAINER_NAME)
         return container
 
+    def _prepare_flask_log_dir(self) -> None:
```

Comment:
Should this be in the flask module?

Reply:
The flask module (`webserver.py`) receives the designated log path from charm, so I think it's the charm's duty to prepare the log path before passing the log path to webserver.

Reply:
I can see why there is a reason to go one way or the other, the way logging works seems to be more application specific than general. That is why it would make more sense for the webserver module to be responsible for setting up the logging directory and anything else the application needs to be able to log. The webserver module can then say to the charm and any other integrations "here is where you can find the logs for the workload". Any thoughts on that?

Reply:
If I put the flask logging directory preparation inside the the webserver module, then the `Observability` class will need to accept a webserver object as argument, is that okay?

Reply:
The webserver state? I think that is exactly the purpose of the state yes

Reply:
Not the state, but the webserver object, since the logging directory is managed by webserver now. It will be something like this

```python
class Observability:  # pylint: disable=too-few-public-methods
    def __init__(self, charm: CharmBase, charm_state: CharmState, webserver: Webserver):
```

Reply:
Could we make the logging directory part of the state so that we just need to pass the webserver state object to `Obervability`?

Reply:
Currently, we are putting the logging directory inside the `CharmState`, but since the charm state is derived from the charm, the preparation of the logging directory (mkdir in workload container) is done by the charm.

Reply:
I moved the preparation of logging directories to the webserver module, thanks!